test/com/google/javascript/jscomp/PolymerPassTest.java

```diff
@@ -109,8 +109,9 @@
           "/** @const */ $jscomp.scope = {};",
           "/**",
           " * @param {!Function} type",
-          " * @param {Object} object",
-          " * @return {Object}",
+          " * @param {T} object",
+          " * @return {T}",
+          " * @template T",
```

Comment:
This seems to just echo back the text of the polyfill.
Is it possible to add some kind of test to show that this actually fixes the problem you're trying to address?

Reply:
Currently the only way I know how to test this is with a full integration test including property renaming. That would be very brittle.

Once the type checker understands ES6 code, the `disableTypeCheck()` calls can be removed in the PolymerPassTest.java file and it should automatically be tested.

Reply:
Just have it return ? then.

Reply:
I think we might lose substantial type checking if we use the unknown type. For instance, the original code would look like:

```js
/**
 * @typedef {{
 * type: !Function,
 * value: *,
 * readOnly: (boolean | undefined),
 * computed: (string | undefined),
 * reflectToAttribute: (boolean | undefined),
 * notify: (boolean | undefined),
 * observer: (string | function(*,*) | undefined)
 * }}
 */
let PolymerElementPropertiesMeta;

/**
 * @typedef {Object<string, !PolymerElementPropertiesMeta>}
 */
let PolymerElementProperties;

class FooElement extends Polymer.Element {
  static get is() { return 'foo-element'; }
  /** @return {!PolymerElementProperties} */
  static get properties() {
    return {
      myProp: {
        type: String,
        value: 'foo'
      }
    };
  }
}
```

The PolymerPass then wraps the anonymous object with the reflection call. I still want type checking on the anonymous object itself, if possible. Don't we keep that with generics?

Reply:
Sounds like I can consider this comment thread as closed with "works as intended" then.
@ChadKillingsworth do you see how to fix the issue I pointed out on `reflectobject.js` line 20?